sdk/eventhubs/azure-messaging-eventhubs/test/eventhubs-stress-test/src/eventhubs_stress_test.cpp

```diff
@@ -72,11 +106,67 @@ constexpr const bool LogDefault = false;
 
 bool LogToConsole{LogDefault};
 
-void InitLogger()
+// Log level textual representation, including space padding, matches slf4j and log4net.
+std::string const ErrorText = "ERROR";
+std::string const WarningText = "WARN ";
+std::string const InformationalText = "INFO ";
+std::string const VerboseText = "DEBUG";
+std::string const UnknownText = "?????";
+
+inline std::string const& LogLevelToConsoleString(Azure::Core::Diagnostics::Logger::Level logLevel)
+{
+  switch (logLevel)
+  {
+    case Azure::Core::Diagnostics::Logger::Level::Error:
```

Comment:
this method has multiple return points , personal preference, if i remember the compiler has mental problems with these cases,  can we have a retValue, set in the switch and returned at the end ? also takes case of the default value by initing it to the default.

Reply:
FWIW, this is a copy of the master in `EnvironmentLogListener.cpp`.

The issue you are referencing is related to the NRVO (named return value optimization) where the compiler doesn't always do a std::move on a returned value. But in this case we're returning a *reference* to a std::string, so there is no move operation happening.

I'll change the code to no longer use `std::string` and make the LogLevelToConsoleString a constexpr function.